File: backend/app/catalogo.py
"""
Catálogo de servicios del lado del servidor.

**Esta es la fuente de verdad del cobro.** El catálogo del front
(`src/app/core/catalogo.ts`) es para mostrar; el dinero se calcula acá, porque
lo que el navegador manda no es confiable.

Los dos tienen que decir lo mismo. Hoy se mantienen a mano, lo que es una
duplicación real y el lugar más probable de una discrepancia entre el precio
que el cliente vio y el que se le cobró. Lo correcto es que el front lea de
`GET /api/catalogo` y esta sea la única copia; queda pendiente.

Formatos: solo los que el motor procesa hoy de punta a punta. Doc_Check
reconoce por magic bytes PDF, imágenes y audio, y nada más. El catálogo del
front ofrece además video, Office y ZIP, que hay que deshabilitar hasta que el
motor los cubra: aceptar un archivo no es procesarlo.
"""

from dataclasses import dataclass


@dataclass(frozen=True)
class Servicio:
    id: str
    nombre: str
    # Cómo se cuenta lo que se cobra: "minutos", "paginas" o "ejecuciones".
    unidad_medida: str
    # Precio por unidad en CLP, sin IVA.
    precio: int
    extensiones: frozenset[str]
    max_mb: int
    # Un servicio sin formatos que el motor procese queda fuera hasta que los
    # tenga. Es preferible decirlo a aceptar el archivo y fallar después.
    disponible: bool = True
    motivo_no_disponible: str = ""


# Tope por documento mientras el archivo viaje en base64 dentro del body. Al
# pasar a Blob Storage sube a lo que promete el catálogo del front.
_MAX_MB = 25

_SERVICIOS: tuple[Servicio, ...] = (
    Servicio(
        id="transcripcion",
        nombre="Transcripción de audio y video",
        unidad_medida="minutos",
        precio=12,
        # Sin video: Doc_Check no reconoce MP4/MOV/MKV, y Speech necesita que
        # se le extraiga el audio antes. Sin .m4a: es un contenedor MP4.
        extensiones=frozenset({".mp3", ".wav", ".ogg", ".flac", ".aac"}),
        max_mb=_MAX_MB,
    ),
    Servicio(
        id="documentos",
        nombre="Análisis de documentos",
        unidad_medida="paginas",
        precio=35,
        extensiones=frozenset({".pdf", ".jpg", ".jpeg", ".png", ".tiff", ".bmp"}),
        max_mb=_MAX_MB,
    ),
    Servicio(
        id="conversaciones",
        nombre="Análisis de conversaciones",
        unidad_medida="minutos",
        precio=25,
        extensiones=frozenset({".mp3", ".wav", ".ogg", ".flac", ".aac"}),
        max_mb=_MAX_MB,
    ),
    Servicio(
        id="automatizacion",
        nombre="Automatización de procesos",
        unidad_medida="ejecuciones",
        precio=18,
        # CSV, XLSX y JSON no pasan Doc_Check, así que el servicio no tiene
        # ningún formato viable todavía.
        extensiones=frozenset(),
        max_mb=_MAX_MB,
        disponible=False,
        motivo_no_disponible="Disponible por API. La carga de archivos desde el portal llega pronto.",
    ),
)

CATALOGO: dict[str, Servicio] = {s.id: s for s in _SERVICIOS}


def servicio_por_id(id_servicio: str) -> Servicio | None:
    return CATALOGO.get(id_servicio)

File: src/app/core/catalogo.ts
import { Servicio, ServicioId } from './modelos';

/**
 * Catálogo de servicios y tarifas. Es la única fuente de precios del sitio:
 * la landing, la página de precios y el panel leen de aquí.
 *
 * Precios en CLP por unidad, sin IVA. Son valores de referencia: ajústalos
 * antes de publicar.
 *
 * OJO: `backend/app/catalogo.py` tiene que decir lo mismo. Esta copia es para
 * mostrar; el cobro se calcula en el servidor, porque lo que manda el
 * navegador no es confiable. Que las dos se separen significa cobrarle al
 * cliente un precio distinto del que vio, así que al tocar precios,
 * extensiones o topes hay que cambiar ambas.
 *
 * `acepta` y `maxMb` describen lo que el motor procesa hoy de punta a punta:
 * audio, PDF e imágenes. Video, Office y ZIP quedan fuera hasta que el motor
 * los cubra — aceptar un archivo no es procesarlo.
 */

/** Tope por archivo mientras viaje en base64 hacia el motor. */
const MAX_MB = 25;

export const CATALOGO: readonly Servicio[] = [
  {
    id: 'transcripcion',
    nombre: 'Transcripción de audio y video',
    icono: 'transcripcion',
    resumen: 'Reuniones, entrevistas y llamadas convertidas en texto con marcas de tiempo.',
    detalle:
      'Sube el archivo y recibe la transcripción con hablantes separados y marcas de tiempo. Reconoce español de Chile, incluidas grabaciones con ruido de fondo o varios participantes.',
    unidad: 'minuto de audio',
    precio: 12,
    incluye: [
      'Separación de hablantes',
      'Marcas de tiempo por párrafo',
      'Exporta a TXT, DOCX y SRT',
    ],
    // Sin video todavía: hay que extraerle el audio antes de transcribirlo.
    acepta: ['.mp3', '.wav', '.ogg', '.flac', '.aac'],
    maxMb: MAX_MB,
    disponible: true,
    motivoNoDisponible: '',
  },
  {
    id: 'documentos',
    nombre: 'Análisis de documentos',
    icono: 'documento',
    resumen: 'Extrae los datos clave de contratos, facturas e informes, y los compara entre sí.',
    detalle:
      'Lee documentos —incluidos los escaneados—, extrae los campos que definas y los cruza para detectar diferencias y faltantes. Cada dato queda citado con documento y página.',
    unidad: 'página',
    precio: 35,
    incluye: [
      'OCR para documentos escaneados',
      'Campos a medida según tu formato',
      'Cada dato citado con su página',
    ],
    acepta: ['.pdf', '.jpg', '.jpeg', '.png', '.tiff', '.bmp'],
    maxMb: MAX_MB,
    disponible: true,
    motivoNoDisponible: '',
  },
  {
    id: 'conversaciones',
    nombre: 'Análisis de conversaciones',
    icono: 'audio',
    resumen: 'Qué se dijo, en qué tono y si se cumplió el guion, sobre el total de tus llamadas.',
    detalle:
      'Sobre el audio ya transcrito, identifica temas, detecta compromisos, mide el tono de la conversación y verifica el cumplimiento de un guion o checklist definido por ti.',
    unidad: 'minuto analizado',
    precio: 25,
    incluye: [
      'Temas y compromisos detectados',
      'Cumplimiento de guion',
      'Resumen ejecutivo por llamada',
    ],
    acepta: ['.mp3', '.wav', '.ogg', '.flac', '.aac'],
    maxMb: MAX_MB,
    disponible: true,
    motivoNoDisponible: '',
  },
  {
    id: 'automatizacion',
    nombre: 'Automatización de procesos',
    icono: 'automatizacion',
    resumen: 'Encadena pasos con IA y conéctalos a los sistemas que ya usas.',
    detalle:
      'Define un flujo —recibir, clasificar, extraer, validar, notificar— y déjalo corriendo. Se conecta por API o correo con tus sistemas actuales, sin migrar nada.',
    unidad: 'ejecución',
    precio: 18,
    incluye: [
      'Flujos con pasos condicionales',
      'Conectores por API y correo',
      'Reintento automático ante fallas',
    ],
    acepta: [],
    maxMb: MAX_MB,
    disponible: false,
    motivoNoDisponible:
      'Disponible por API. La carga de archivos desde el portal llega pronto.',
  },
];

/** Acceso puntual por id, para pintar el servicio de una solicitud. */
export function servicioPorId(id: ServicioId): Servicio {
  const servicio = CATALOGO.find((s) => s.id === id);
  if (!servicio) {
    throw new Error(`Servicio desconocido: ${id}`);
  }
  return servicio;
}

/** Los que aceptan carga desde el portal, que es lo que ofrece /enviar. */
export const SERVICIOS_DISPONIBLES: readonly Servicio[] = CATALOGO.filter((s) => s.disponible);
